binderhub/launcher.py

```diff
@@ -172,6 +177,19 @@ def unique_name_from_repo(self, repo_url):
             user_data = await self.get_user_data(username)
             if server_name in user_data['servers']:
                 raise web.HTTPError(409, "User %s already has a running server." % username)
+        elif self.named_server_limit_per_user > 0:
+            # authentication is enabled with named servers
+            # check if user has already reached to the limit of named servers
+            user_data = await self.get_user_data(username)
+            len_named_spawners = len([s for s in user_data['servers'] if s != ''])
```

Comment:
For my education: when does it happen that a list item is an empty string here? Was expecting `user_data['servers']` to contain the list of server names, which should always be a none empty string.

```suggestion
            len_named_spawners = len([s for s in user_data['servers'] if s])
```
For even more Pythonicness (in a follow up PR if you  like)

Reply:
> For my education: when does it happen that a list item is an empty string here?

Default server's name is an empty string by default. In auth bhub with named servers, a user can always go to control panel and do "Start My Server" and this is also included in `user_data['servers']`.